x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorScheduling.java

```diff
@@ -0,0 +1,248 @@
+
+    public static class ScheduleConfig implements Writeable, ToXContentObject {
+        private final boolean enabled;
+        private final String interval;
+
+        private static final ParseField ENABLED_FIELD = new ParseField("enabled");
+        private static final ParseField INTERVAL_FIELD = new ParseField("interval");
+
+        /**
+         * @param enabled  flag to disable/enable scheduling
+         * @param interval CRON expression representing the sync schedule
+         */
+        private ScheduleConfig(boolean enabled, String interval) {
+            this.enabled = enabled;
+            this.interval = Objects.requireNonNull(interval, INTERVAL_FIELD.getPreferredName());
+        }
+
+        public ScheduleConfig(StreamInput in) throws IOException {
+            this.enabled = in.readBoolean();
+            this.interval = in.readString();
+        }
+
+        private static final ConstructingObjectParser<ScheduleConfig, Void> PARSER = new ConstructingObjectParser<>(
+            "schedule_config",
+            true,
+            args -> new Builder().setEnabled((boolean) args[0]).setInterval((String) args[1]).build()
+        );
+
+        static {
+            PARSER.declareBoolean(constructorArg(), ENABLED_FIELD);
+            PARSER.declareString(constructorArg(), INTERVAL_FIELD);
+        }
+
+        public static ScheduleConfig fromXContent(XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
+        }
+
+        public static ConstructingObjectParser<ScheduleConfig, Void> getParser() {
+            return PARSER;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field(ENABLED_FIELD.getPreferredName(), enabled);
+                builder.field(INTERVAL_FIELD.getPreferredName(), interval);
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeBoolean(enabled);
+            out.writeString(interval);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            ScheduleConfig that = (ScheduleConfig) o;
+            return enabled == that.enabled && Objects.equals(interval, that.interval);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(enabled, interval);
+        }
+
+        public static class Builder {
+
+            private boolean enabled;
+            private String interval;
+
+            public Builder setEnabled(boolean enabled) {
+                this.enabled = enabled;
+                return this;
+            }
+
+            public Builder setInterval(String interval) {
```

Comment:
Do we want to do any validation on proper cron syntax here?

Reply:
Good call, I found [Cron](https://github.com/elastic/elasticsearch/blob/61191b880c7bbcdac6ac8f01a544ca3abe3385a1/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/scheduler/Cron.java#L200) class in ES which does this already. I will use it and it will handle the validation for us